docs/kratos/self-service/hooks.mdx

```diff
@@ -212,10 +216,57 @@ password: My-Pass-Value
 
 All properties are mandatory.
 
+### Interruptable Web-Hooks
+
+Sometimes notification-like web hooks are not enough and one need to integrate
+some external business logic into user's self-service flows. Here interruptable
+web-hooks comes with help. Basic flow is to pass self-service flow to an
+external service that will decide if the process should continue or user should
+be notified about an error (or fail silently). To achieve that interruptable
+web-hook can be defined by specifying `can_interrupt` flag and pointing it to a
+web service that will process the payload from the webhook and will return a
+predefined JSON that will instruct Kratos to forward this information to the
+front end. When hook service returns status code that is equal or lower than 399
+the self-service flow continues normally but if web hook returns 400 or higher
+Kratos parses the returned JSON and looks for a messages that can be forwarded
+to the front end. Sample payload may look like this:
+
+```json
+{
+  "messages": [
+    {
+      "instance_ptr": "#/traits/foo/bar",
+      "message": "bar field is violating our policies",
+      "detailed_meesages": [
+        {
+          "id": 123,
+          "text": "field must be longer than 12 characters",
+          "type": "validation",
+          "context": {
+            "value": "short value"
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+Since the whole idea of blocking hooks is to allow to interrupt selfservice flow
+and provide user with a meaningfull information (messages/errors) please cofigure
+them in `after` web hooks configurations.
+
+Some clarifications about meaning of some fields:
+
+- `instance_ptr` points to the field in the payload/schema that this error
+  refers to (displayed by the frontend next to the corresponding field)
+- `id` is the unique numeric error ID that helps to translate messages in the
+  front end aplpication.
+
 ## Login
 
-Hooks running before or after successful user login are defined per Self-Service
-Registration Method in Ory Kratos' configuration file.
+For each Registration Method in Ory Kratos' configuration file allows to setup
```

Comment:
I'm not sure what you mean by this sentence, compared to the old version.

Reply:
That looks like an editing error. Reverting. Done in 0721a0c7